### PR TITLE
Disable import button if no selected browser

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/28788

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Steps from issue

### CODE REVIEW

**Review by nullhook — CHANGES_REQUESTED** [review]

File: components/brave_welcome_ui/components/select-browser/style.ts [hunk 1]
```diff
@@ -101,4 +101,24 @@ export const ActionBox = styled.div`
   button {
     color: white;
   }
+
+  button:nth-child(2) {
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_welcome_ui/components/select-browser/style.ts:105]: Took it from figma as you recommended

File: components/brave_welcome_ui/components/select-browser/index.tsx [hunk 2]
```diff
@@ -147,13 +149,14 @@ function SelectBrowser () {
...
-        <Button
-          isPrimary={true}
-          onClick={handleImport}
-          scale="jumbo"
-        >
-          {getLocale('braveWelcomeImportButtonLabel')}
-        </Button>
+          <Button
+            isPrimary={true}
+            isDisabled={!hasSelectedBrowser}
```
[the remark this answers is not in the document]
Comment (nullhook) [reply components/brave_welcome_ui/components/select-browser/index.tsx:154]: Is this a default selection and not user selected?

[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_welcome_ui/components/select-browser/index.tsx:154]: This is taken from the system, user may have not supported browsers

File: components/brave_welcome_ui/components/select-browser/style.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nullhook) [reply components/brave_welcome_ui/components/select-browser/style.ts:105]: Minimize styles, only add necessary props for overriding. Test for desired outcome. Disabled button style should have translucent background, potentially change color. Border radius and font styles should already be in base.

You can perhaps start with below:

```css
background: rgba(255, 255, 255, 0.14);
backdrop-filter: blur(8px);
color: rgba(255, 255, 255, 0.32);
```

[the remark this answers is not in the document]
Comment (nullhook) [reply components/brave_welcome_ui/components/select-browser/style.ts:105]: Do you really need min-height, display flex, or font styles?

[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_welcome_ui/components/select-browser/style.ts:105]: ok, removed, looks good, thanks

**Review by nullhook — APPROVED** [review]
Looks good!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.52.12 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
OS | Windows 11 Version 22H2 (Build 22621.1413)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/28788#issue-1601811480, ensured the following:

* ensured that `Import` is not clickable when a browser has not been selected during the `Import` phase
* ensured that selecting a browser enables the `Import` button without any issues
  * ensured that clicking the `Import` button works as expected and moves onto the next phase of `Import`

**`Example`** | **`Example`**
----------------|-----------------
example1 | example2

**Pull request merged**